src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1885,14 +1885,19 @@ internal override IConvertibleConversion ClassifyConvertibleConversion(IOperatio
         /// Returns a new ArrayTypeSymbol representing an array type tied to the base types of the
         /// COR Library in this Compilation.
         /// </summary>
-        internal ArrayTypeSymbol CreateArrayTypeSymbol(TypeSymbol elementType, int rank = 1)
+        internal ArrayTypeSymbol CreateArrayTypeSymbol(TypeSymbol elementType, int rank = 1, NullableAnnotation elementNullableAnnotation = NullableAnnotation.Oblivious)
         {
             if ((object)elementType == null)
             {
                 throw new ArgumentNullException(nameof(elementType));
             }
 
-            return ArrayTypeSymbol.CreateCSharpArray(this.Assembly, TypeWithAnnotations.Create(elementType), rank);
+            if (rank < 1)
```

Comment:
This is technically a breaking change. I'm fine with it, especially as if you drill down the stack we assert that `rank >= 1`, but I just wanted to call out that it is a behavioral break.

Reply:
What was the previous behavior for rank = 0?

---
In reply to: [302305826](https://github.com/dotnet/roslyn/pull/37048#discussion_r302305826) [](ancestors = 302305826)

Reply:
Previously we created an `MDArrayNoSizesNoBounds` with  `rank == 0`. (In `DEBUG` builds, we assert `rank != 0`.)

---
In reply to: [302726173](https://github.com/dotnet/roslyn/pull/37048#discussion_r302726173) [](ancestors = 302726173,302305826)